src/transformers/models/clip/configuration_clip.py

```diff
@@ -330,11 +330,26 @@ def __init__(
                 text_config = {}
 
             # This is the complete result when using `text_config_dict`.
-            _text_config_dict = CLIPTextConfig(**text_config_dict).to_dict()
+            # We can't simply use `to_dict` due to the type of keys in `text_config_dict["id2label"]` being `int`
+            # instead of `str`.
+            _text_config_dict = json.loads(CLIPTextConfig(**text_config_dict).to_json_string(use_diff=False))
```

Comment:
This is for `id2label`

Reply:
Sorry could you explain this in a bit more detail. Why do we need to take `"id2label"` into account here?

Reply:
- `id2label` in the line `_text_config_dict = CLIPTextConfig(**text_config_dict).to_dict()`:
  - `CLIPTextConfig(**text_config_dict)` is a config object, and its `id2label` has `int` (`0`, `1`, etc) as keys, see [here](https://github.com/huggingface/transformers/blob/71d47f0ad498b7649f11d3a9cca3cd3585e4341f/src/transformers/configuration_utils.py#L332)
  - `to_dict` will keep `id2label` untouched: still has `int` (`0`, `1`, etc) as keys

- `text_config` argument is loaded from the `text_config` attribute in a config json file --> it is a dictionary (not an config object) at this sage, and its `id2label` is also a dictionary with `str` (`"0"`, `"1"`) as keys.

Reply:
we have to make them having the same format, otherwise their `id2label` are different and the warning would be given